### PR TITLE
Feature/sql logic

### PR DESCRIPTION
**Pull request by fgbogdan**
Hi, 
we use your plugin in our app - and we have added some logic in CH sql before sending it to CH.
The sqlquery is processed before to be sent to CH, using some minimal javascript conditional syntax as:
```
<%if(($to - $from) / 3600 > 3){%> 
   SELECT something FROM ...
<%}else{%>
  SELECT anotherThing FROM ...
<%}%>
```

The available keywords are:
var|if|for|else|switch|case|break

Please let me know if you need more details or samples.
Thanks for your work till now !

### CODE REVIEW

**Comment by Slach** [comment]
Hello @fgbogdan, sorry for late resonse, 
could you resolve conflicts
and add documentation to README with examples
and also add jest unit tests?

**Comment by fgbogdan** [comment]
Hi, Thanks for your answer.
I have added details in README.MD.
The conflicts where generated by dist modules - and these are generated on each compile. (npm build).
For Junit - i have no idea how to, but I will try.

File: src/scanner.ts [hunk 1]
```diff
@@ -55,15 +55,22 @@ export default class Scanner {
...
-            return false;
+          return false;
         }
-        let r = this.re.exec(this._s);
-        if (r === null) {
-            throw("cannot find next token in [" + this._s + "]");
+        // if addnotation
+        if (this._s.startsWith("<%")) {
+          let nPos = this._s.search("%>");
+          this._s = this._s.substring(nPos + 2);
```
Comment (Slach) [inline src/scanner.ts:63]: I think we need to add throw exception if closed template brackets don't found here
Also, i think we need add some escape mechanism
when we need use <% or %> inside queries for queries like
```
SELECT $timeSeries, '<%hostname%>' AS label, sum(gauge) FROM $table WHERE $timeFilter
```

Comment (fgbogdan) [reply src/scanner.ts:63]: Correct. Will inform you when it is done.

**Comment by Slach** [comment]
@fgbogdan look test cases on https://github.com/Vertamedia/clickhouse-grafana/blob/master/spec/scanner_specs.jest.ts and add test cases for your render() function

**Comment by fgbogdan** [comment]
Hi again, 

I have added the test for matching <%.%> and also some tests in scanner_specs.jest.ts
For the escape mechanism - will think about.

**Comment by fgbogdan** [comment]
Hi Slach, 
should I do something else to fulfill the case ?
Still no idea for "escape mechanism" when the tables or fields contains themselves <% or %>, excepting some dummy escape chars like /<% or <<%.
Any idea will be appreciate.
10x
Bogdan

**Comment by Slach** [comment]
@fgbogdan  could you run `docker-compose run frontend_builder` and `docker-compose run backend_builder` and check render function works with alert functionality?

could you add some example dashboards with SQL logic and alerts to `docker/grafana/dashboards/`?

**Comment by Slach** [comment]
oops sorry, it was miss click for close PR

**Comment by fgbogdan** [comment]
Hi, ok - will try that next week.

**Comment by Slach** [comment]
will wait, thanks a lot for contributing

**Comment by fgbogdan** [comment]
Hi, got some time this morning and added the dash with sql logic and alert. If you want something more fancy just let me know

**Comment by Slach** [comment]
@fgbogdan thanks for contributing

**Comment by Slach** [comment]
sorry @fgbogdan this functionality make this plugin to available XSS attacks, I need revert it ASAP

**Pull request merged**